core/state/state_object.go

```diff
@@ -72,11 +71,6 @@ func (s Storage) Copy() Storage {
 // - Account values as well as storages can be accessed and modified through the object.
 // - Finally, call commit to return the changes of storage trie and update account data.
 type stateObject struct {
-	// dataLock protects the [data] field to prevent a race condition
-	// in the transaction pool tests. TODO remove after re-implementing
-	// tx pool to be synchronous.
-	dataLock sync.RWMutex
```

Comment:
Are we sure this is safe to remove?

Reply:
I think these were fixed way before with https://github.com/ava-labs/coreth/commit/8e32d4371af9e7e8d12c64451bf0dcefc14d4e88#diff-64a06d0cea30e12d6001fb322e583930c1cf3879bca4e8019fa89499ed9f7de4

Also seems like this was only an issue in tests. cc @aaronbuchwald

Reply:
I also added `-race` flag to UTs to catch these.